app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+                $subkws .= $value->keyword . ',';
+            } elseif ($value->type == 1) {
+                $mainkw .= $value->keyword . ',';
+            }
+        }
+        $action_delete = $this->currentUser->is(User::ADMIN) ? 'Webapp\PostController@getDelete' : 'Webapp\PostController@getWriterDelete';
+
+        $clientChecksum = $post->getLastClientChecksum();
+        $recoverChecksum = null;
+
+        if(!empty($checksum) && $post->getChecksum() != $clientChecksum) {
+            $this->_flashMessage($request,
+                'Content failed integrity check, maybe error when saving or database was manipulated. '.
+                'Editor will try to recover from backup. If error continue, please contact admin with message: Checksum failed('.$checksum.')'
+                , 'danger');
+            $recoverChecksum = $clientChecksum;
+        }
+
+        $site = Site::find($post->site_id);
+        $additionCss = (!empty($site)) ? $site->getAdditionCss() :null;
+        $draftUnsaved = PostHistory::getFirstFrom($post, PostHistory::STATUS_DRAFT, $post->updated_at->toDateTimeString());
+
+        return view('webapp/posts/edit'.(($readOnly) ? '_readonly' : ''), [
+            'post' => $post,
+            'lockStatus' => $lockStatus,
+            'sites' => Site::where('status', 1)->get(),
+            'categoriesSet' => $categoriesSet,
+            'subkws' => $subkws,
+            'mainkw' => $mainkw,
+            'action_delete' => $action_delete,
+            'additionCss' => $additionCss,
+            'postChecksum' => $clientChecksum,
+            'recoverChecksum' => $recoverChecksum,
+            'draftUnsaved' => $draftUnsaved,
+            'readOnly' => $readOnly
+        ]);
+    }
+
+    public function postEdit(Request $request, $postId = 0) {
+        if(empty($post = Post::find($postId)) || !$post->canEditBy($this->currentUser) ) {
+            return '404 Not Found';
+        }
+
+
+        if($post->isLock($this->currentSession)) {
+            $this->_flashMessage($request, 'このコンテンツは現在ロックされています。編集できません。', 'danger');
+            return redirect()->to(action('Webapp\\PostController@getEdit', $postId))->send();
+        }
+        if($request->get('site_id') == ''){
+            $this->_flashMessage($request, 'ターゲットサイトを選んでください', 'danger');
+            return redirect()->to(action('Webapp\\PostController@getEdit', $postId))->send();
+        }
+
+        $postTitle = $request->get('post_title');
+        $postContent = $request->get('post_content');
+        $postDescription = trim($request->get('post_description'));
+
+        $previousContent = $post->content;
+        $post->title = !empty($postTitle) ? $postTitle : 'タイトルなし';
+        $post->content = $postContent;
+        $post->category = $request->get('category');
+        $post->description = $postDescription;
+        $post->feature_img = !empty($request->get('new_feature_img_url')) ? $request->get('new_feature_img_url') : $request->get('feature_img_url');
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "new_feature_img_url" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1192)